mmdet/core/hook/neptunelogger_hook.py

```diff
@@ -0,0 +1,215 @@
+
+        self.interval = interval
+
+        self.log_model = log_model
+        self.log_checkpoint = log_checkpoint
+        self.log_model_diagram = log_model_diagram
+
+        self.num_eval_predictions = num_eval_predictions
+        self.log_eval_predictions = (num_eval_predictions > 0)
+
+        self.ckpt_hook: CheckpointHook = None
+        self.ckpt_interval: int = None
+
+        self.eval_hook: EvalHook = None
+        self.eval_interval: int = None
+
+        self.val_dataset = None
+
+        self.kwargs = kwargs
+
+    def _log_integration_version(self) -> None:
+        self._run[self.INTEGRATION_VERSION_KEY] = version.__version__
+
+    def _log_config(self, runner) -> None:
+        if runner.meta is not None and runner.meta.get('exp_name',
+                                                       None) is not None:
+            src_cfg_path = osp.join(runner.work_dir,
+                                    runner.meta.get('exp_name', None))
+            config = Config.fromfile(src_cfg_path)
+            if osp.exists(src_cfg_path):
+                self.base_handler['config'] = config.pretty_text
+        else:
+            runner.logger.warning('No meta information found in the runner. ')
+
+    @master_only
+    def before_run(self, runner) -> None:
+        """Logs config if exists, inspects the hooks in search of checkpointing
+        and evaluation hooks.
+
+        Raises a warning if checkpointing is enabled, but the dedicated hook is
+        not present. Raises a warning if evaluation logging is enabled, but the
+        dedicated hook is not present.
+        """
+        self._log_integration_version()
+        self._log_config(runner)
+
+        # Inspect CheckpointHook and EvalHook
+        for hook in runner.hooks:
+            if isinstance(hook, CheckpointHook):
+                self.ckpt_hook = hook
+            if isinstance(hook, (EvalHook, DistEvalHook)):
+                self.eval_hook = hook
+
+        if self.log_checkpoint:
+            if self.ckpt_hook is None:
+                self.log_checkpoint = False
+                runner.logger.warning(
+                    'To log checkpoint in NeptuneHook, `CheckpointHook` is'
+                    'required, please check hooks in the runner.')
+            else:
+                self.ckpt_interval = self.ckpt_hook.interval
+
+        if self.log_eval_predictions:
+            if self.eval_hook is None:
+                self.log_eval_predictions = False
+                runner.logger.warning(
+                    'To log evaluation NeptuneHook, '
+                    '`EvalHook` or `DistEvalHook` in mmdet '
+                    'is required, please check whether the validation '
+                    'is enabled.')
+            else:
+                self.eval_interval = self.eval_hook.interval
+                self.val_dataset = self.eval_hook.dataloader.dataset
+
+    def _log_buffer(self, runner, category, log_eval=True) -> None:
```

Comment:
I think the name should be updated to reflect the fact that it also does evaluation. Wdyt?